NuRadioMC/examples/06_webinar/W02RunSimulation.py

```diff
@@ -85,10 +85,12 @@ def _detector_simulation(self):
         equivalently, a time step of 0.2 ns. Such a high resolution is not common
         at all in radio experiments, where the sampling rates tend to lie around
         the gigahertz. So, if we want our simulation as realistic as possible,
-        we must resample to lower sampling rates. Let us a assume we have a
-        sampling rate of 2 GHz. Then, we can use the channelResampler to downsample.
+        we must resample to lower sampling rates, to the actual sampling rate of
+        our analog-to-digital converter. We have specified in our detector.json
+        file an adc_sampling_frequency of 2 GHz, which can be accessed using the
+        property _sampling_rate_detector.
         """
-        new_sampling_rate = 2 * units.GHz
+        new_sampling_rate = self._sampling_rate_detector
```

Comment:
for most detectors this is still a bad choice. Rather use the internal sampling rate of the simulation

Reply:
Why? The simulation runs with 5 GHz. No current detector does that...

Reply:
an analog trigger has no sampling problem. It always sees the full signal amplitude, i.e. what we get with infinite upsampling.

Reply:
and before saving the nur files, the traces get downsampled to detector sampling rate automatically.

Reply:
so to resemble the detector situation, you need to run the trigger on an upsampled trace

Reply:
That is if the trigger is analog. For the phased array, for instance, the trigger has to be digital. I can remove the resampling for the example - my only intention was being pedagogical and teach how to use a feature I always needed.

Reply:
Ok I see your point. If this is described in the documentation/comments choosing the detector sampling rate is fine too I guess. But doesn't the phased array use an internal upsampling and anyway runs on a very high sampling rate of 3.7GHz?

Reply:
Not exactly. The ideal phased array for RNO had 3 GS/s, but that is not implementable due to power budget restrictions. What I'm studying right now with the alias branch is the possibility of using low sampling rates and then upsampling to 1 GS/s with a FIR via the FPGA. It kind of looks ok, but 1 GS/s doesn't give a lot of angles to work with, since the delays are more limited than with 3 GS/s, so the effective volume is going to drop.

Reply:
shouldn't this up/downsampling be directly implemented into the trigger module? Otherwise the trigger performance will depend on the users choice of internal sampling rate etc. So a simulation of the PA is error prone.

Reply:
The resampling needed to be done in the steering file so far if there was no ADC conversion, but that's no longer the case with the new ADC module, everything is read from the detector file. I don't find that cumbersome to modify the channel sampling rate in the steering file no more than adding noise and filters manually in the same file, but that's my opinion.

Reply:
And I wouldn't say it's error-prone. The original phased array module works as it should with every sampling rate, you just need to give the right one at some point, whether in the detector file or the steering file.

Reply:
To agree with your documentation, you should add 
`channelResampler.run(self._evt, self._station, self._det, sampling_rate=1/self._dt)`
This is how the internal logic with the internal sampling rate works. Or actually better set `new_sampling_rate` to `1/self._dt` so that it is consistent with the adding noise part. 

Cool great that it is incorporated in the ADC module.

Reply:
But that's not what I would want to do with a digital trigger. 1/self._dt is 5 GHz, and I would like to show the user how to downsample to 2 GHz. What am I not getting here?

Reply:
But you are simulating the high/low trigger which is an analog trigger. So to simulate it properly, you must oversample the trace. If you call this module with the detector sampling rate you get wrong results (too few triggered events).

Reply:
Changed. Take a look when you can.